src/Api/Concern/DecodesAddressFields.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\Api\Concern;
+
+use MyParcelNL\Pdk\Base\Support\Utils;
+
+trait DecodesAddressFields
+{
+    /**
+     * @param  null|array $input
+     *
+     * @return null|array
+     */
+    protected function decodeAddress(?array $input): ?array
+    {
+        $data = Utils::changeArrayKeysCase($this->filter($input) ?? []);
+
+        if (isset($data['street']) || isset($data['number'])) {
+            $data['address1'] = trim(
+                implode(' ', [
+                    $data['street'],
+                    $data['number'],
+                    $data['numberSuffix'] ?? $data['boxNumber'] ?? '',
+                ])
+            );
+
+            unset($data['street'], $data['number'], $data['numberSuffix'], $data['boxNumber']);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param  null|array $array
+     *
+     * @return array
+     * @see \MyParcelNL\Pdk\App\Order\Model\ShippingAddress
+     */
+    protected function decodeAddress2(?array $array): array
```

Comment:
Zou dit een betere naam moeten krijgen? Of is er ergens een `Address2` in deze vorm?

Reply:
ah scherp, deze kon weg!